src/Microsoft.IdentityModel.Tokens/RsaSecurityKey.cs

```diff
@@ -27,69 +27,79 @@
 
 using System;
 using System.Security.Cryptography;
+using Microsoft.IdentityModel.Logging;
 
 namespace Microsoft.IdentityModel.Tokens
 {
     public class RsaSecurityKey : AsymmetricSecurityKey
     {
-        private RSAParameters _rsaParamaeters;
-
         public RsaSecurityKey(RSAParameters rsaParameters)
         {
             // must have private or public key
-            bool hasPrivateKey = rsaParameters.D != null && rsaParameters.DP != null && rsaParameters.DQ != null && rsaParameters.P != null && rsaParameters.Q != null;
-            bool hasPublicKey = rsaParameters.Exponent != null && rsaParameters.Modulus != null;
-            if (!hasPrivateKey && !hasPublicKey)
+            HasPrivateKey = rsaParameters.D != null && rsaParameters.DP != null && rsaParameters.DQ != null && rsaParameters.P != null && rsaParameters.Q != null;
+            HasPublicKey = rsaParameters.Exponent != null && rsaParameters.Modulus != null;
+            if (!HasPrivateKey && !HasPublicKey)
             {
-                throw new ArgumentException("no public or private key material found");
+                throw LogHelper.LogException<ArgumentException>("No public or private key material found");
             }
 
-            _rsaParamaeters = rsaParameters;
+            Parameters = rsaParameters;
         }
 
-        public override int KeySize
+        public RsaSecurityKey(RSA rsa)
```

Comment:
Nothing, which sucks, terribly.
If you really want to keep the `RsaParameters` property for conveniency, it might be better to make a simple wrapper calling `Rsa.ExportParameters`.